internal/campaigns/volume_workspace.go

```diff
@@ -3,29 +3,44 @@ package campaigns
 import (
 	"bytes"
 	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
 	"io/ioutil"
 	"os"
 
 	"github.com/pkg/errors"
 
+	"github.com/sourcegraph/src-cli/internal/campaigns/docker"
 	"github.com/sourcegraph/src-cli/internal/campaigns/graphql"
 	"github.com/sourcegraph/src-cli/internal/exec"
 	"github.com/sourcegraph/src-cli/internal/version"
 )
 
-type dockerVolumeWorkspaceCreator struct {
-	tempDir string
-}
+type dockerVolumeWorkspaceCreator struct{ tempDir string }
 
 var _ WorkspaceCreator = &dockerVolumeWorkspaceCreator{}
 
-func (wc *dockerVolumeWorkspaceCreator) Create(ctx context.Context, repo *graphql.Repository, zip string) (Workspace, error) {
+func (wc *dockerVolumeWorkspaceCreator) Create(ctx context.Context, repo *graphql.Repository, steps []Step, zip string) (Workspace, error) {
 	volume, err := wc.createVolume(ctx)
 	if err != nil {
 		return nil, errors.Wrap(err, "creating Docker volume")
 	}
 
-	w := &dockerVolumeWorkspace{tempDir: wc.tempDir, volume: volume}
+	// Figure out the user that containers will be run as.
+	ug := docker.UIDGID{}
+	if len(steps) > 0 {
```

Comment:
Mental note: we should probably ensure somewhere that we don't download any archives and/or do things if we have zero steps ;)

Reply:
Agreed!